src/components/notes/editor.jsx

```diff
@@ -13,196 +7,58 @@ import EditorLoading from 'components/notes/editor-loading'
-    title && title != '' ? `${appFullName} - ${title}` : appFullName
-}
-
-async function loadNote(
-  serviceClient,
-  noteId,
-  loading,
-  setLoading,
-  setDoc,
-  setTitle
-) {
-  try {
-    if (!loading) {
-      setLoading(true)
-    }
-    const doc = await serviceClient.getDoc(noteId)
-    setTitle(doc.title || '')
-    setDoc(doc)
-  } catch (e) {
-    setTitle(false)
-    setDoc(false)
-  }
-  setLoading(false)
-}
-
-function getLocalTitleChangeCallback(serviceClient, noteId, title, setTitle) {
-  return e => {
-    const newTitle = e.target.value
-    const modifiedTitle = newTitle
-    if (title != modifiedTitle) {
-      setTitle(modifiedTitle)
-      serviceClient.setTitle(noteId, modifiedTitle)
-    }
-  }
-}
-
 const Editor = translate()(
   withClient(function(props) {
-    const { client, noteId, t } = props
-    const userName = useMemo(
-      () => props.userName || getShortNameFromClient(client),
-      [props.userName]
-    )
-    const appFullName = useMemo(getAppFullName)
-    const isPublic = useContext(IsPublicContext)
-
-    // alias for later shortcuts
-    const docId = noteId
-    const userId = userName
-    const cozyClient = client
-
-    // state
-    const [loading, setLoading] = useState(true)
-    const [doc, setDoc] = useState(undefined)
-    const [title, setTitle] = useState(undefined)
+    // base parameters
+    const { client: cozyClient, noteId, t } = props
 
     // plugins and config
-    const serviceClient = useMemo(
-      () => {
-        return new ServiceClient({ userId, userName, cozyClient })
-      },
-      [noteId]
-    )
-    const docVersion = doc && doc.version
-    //console.log("docVersion", doc, doc && doc.version, docVersion)
-    const collabProvider = useMemo(
-      () => {
-        //console.log("collab provider memo", docVersion)
-        if (docVersion !== undefined) {
-          //console.log('new collabProvider')
-          const provider = new CollabProvider(
-            { version: doc.version, docId },
-            serviceClient
-          )
-          // The following object is defined in an Atlassian API.
-          // `provider` expects a Promise, even if we wouldn't
-          //  need it ourselves.
-          return {
-            useNativePlugin: true,
-            provider: Promise.resolve(provider),
-            inviteToEditHandler: () => undefined,
-            isInviteToEditButtonSelected: false,
-            userId: serviceClient.getSessionId()
-          }
-        } else {
-          return null
-        }
-      },
-      [noteId, docVersion, userName, serviceClient]
-    )
-    //console.log("end collabProviderMemo", collabProvider)
-
-    // fetch the actual note on load
-    useEffect(
-      () => {
-        loadNote(serviceClient, noteId, loading, setLoading, setDoc, setTitle)
-      },
-      [noteId]
-    )
+    const isPublic = useContext(IsPublicContext)
+    const { userName, userId } = useUser({
+      userName: props.userName,
```

Comment:
Nit, but might as well destructure `userName ` from `props` at the start with the other props. It makes it explicit what props are being used.

Reply:
I have that pattern two or three times in notes. I welcome suggestions on what to to.

Not destructuring allows me here to use `userName` everywhere whitout having one variable for the input and one for the output (and be sure that if I use `userName`, I'm sure it is a normalized value and not some raw thing that I shouldn't use)

Reply:
Aaah I see, that's fair enough. I don't have a better idea on how to handle this at the moment, with the explanation I also think not destructuring is better in this case.

Reply:
One solution would be to use `const { userName: propUserName } = props` ?